### PR TITLE
fix: allow escaping spaces in words

### PR DESCRIPTION
**Pull request by dsherret**
I was investigating adding glob support and was thinking about file paths with spaces in the name. That's when I realized this functionality was broken. See the tests for examples of what's now fixed. It's not perfect still... just an incremental improvement.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -547,7 +553,22 @@ pub(crate) fn parse_string_parts(
           "Back ticks in strings is currently not supported.",
         )
       },
-      map(if_true(next_char, |c| allow_char(*c)), PendingPart::Char),
+      // words can have escaped spaces
+      map(
+        if_true(preceded(char('\\'), char(' ')), |_| {
```
Comment (bartlomieju) [inline src/parser.rs:558]: https://imgflip.com/i/697idw

**Review by bartlomieju — APPROVED** [review]
I must say the code is confusing, but looking at the tests it's pretty obvious what's being fixed. LGTM!

File: src/parser.rs (hunk 1 above)
Comment (dsherret) [reply src/parser.rs:558]: Will do in a follow up PR lol

**Comment by dsherret** [comment]
> I must say the code is confusing

Yeah, I thought the same thing too. I'll see about improving it when adding glob support.

**Pull request merged**